### ISSUE TITLE
Configuration not honoring environment variable k/v pairs

### ISSUE THREAD

**Issue by Neutrollized** [cantaloupe-project/cantaloupe#523]
Greetings!

(TL;DR available at the bottom)

Lately I've been playing around with the Cantaloupe server and was looking to run a containerized version of it.  I built a few different images based off of multiple Dockerfile examples in various GitHub repos.  In any case, during my testing, my setup involves enabling the ***/admin*** endpoint as well as setting the Source to use AWS S3, providing ACCESS_KEY_ID, SECRET_KEY, and S3 Bucket name.  It works as expected when I define my settings in the *cantaloupe.properties* file.  However, when I try to define my settings by passing them in as [environment variables](https://cantaloupe-project.github.io/manual/4.1/configuration.html), it doesn't get passed/recognized as expected.  I did double-check my spelling as well verify that the env vars *are* set within the container itself.

The only thing I seem to be able to toggle is the `ENDPOINT_ADMIN_ENABLED` (**endpoint.admin.enabled**) and set the corresponding password with `ENDPOINT_ADMIN_SECRET` (**endpoint.admin.secret**)

I have tested and able to reproduce the same bug(?) for in both on my Mac running the containerized version as well as from my Raspberry Pi (I have tried versions: `v4.1.5`, `v4.1.9` and `v5.0.3`)

I've also tried running the docker-compose examples from [MITLibraries/docker-cantaloupe](https://github.com/MITLibraries/docker-cantaloupe), but it too does not recognize any of the environment variable settings that were set within `docker-compose.yaml`.  (I did have to point the GitHub project from **medusa-project** to **cantaloupe-project**)

### How to reproduce
Set the following env vars:
- `export ENDPOINT_ADMIN_ENABLED=true`
- `export ENDPOINT_ADMIN_SECRET=myS3cr3t`
- `export ENDPOINT_IIIF_1_ENABLED=true`
- `export ENDPOINT_IIIF_2_ENABLED=false`
- `export SOURCE_STATIC=S3Source`

Then run the Cantaloupe server (i.e. `java -Dcantaloupe.config=cantaloupe.properties.sample -Xmx2g -jar ./cantaloupe-4.1.5.war`)

The default setting for **endpoint.admin.enabled** should be **false**, so this will override that and set a password for you to login with.  After you login, however, you will see that none of the other settings you specified actually got honored (1.x endpoint still disabled, 2.x endpoint still enabled, Static Source still showing FilesystemSource)

### TL;DR
- configuration does not recognize environment variables as a valid configuration method (except for toggling the `/admin` endpoint and setting the secret)
- above behavior consistent in both containerized images of Cantaloupe as well as one that runs on a more standard Linux server (Raspberry Pi)

Thanks for reading!

Cheers,
Glen

**Comment by adolski** [comment]
Hi @Neutrollized ,

I can't reproduce the issue with environment variables not working--these work for me no matter which ones I set. 

I can say, though, that the control panel is not really meant to be used simultaneously with environment variable config. The control panel is basically meant as a GUI config file editor. Since the environment is read-only to the Java process, it's assumed that if you are setting environment variables, you are already comfortable with what they are set to, and don't require runtime tweaking. Also, when using the control panel with environment config, it's not safe to assume that it will show you the correct state of any given config key.

**Comment by Neutrollized** [comment]
Hi @adolski ,

Thanks for the quick response.  You're absolutely correct -- it appears that the admin control panel does not accurate reflect the actual settings set via env vars, which was what was throwing me off as a newbie, I was relying on that to validate settings, etc. and since it correctly reflects what's in the cantaloupe.properties, I assumed it would do the same for env vars as well. 

Thanks!
